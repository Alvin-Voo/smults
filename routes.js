const routes = require('next-routes')(); //this is a function

routes
.add('/search','index')
.add('/about','about');

module.exports = routes;
