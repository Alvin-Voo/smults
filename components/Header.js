import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Menu, Message } from 'semantic-ui-react';
import { Router, Link } from '../routes';

class Header extends Component {
  state = { activeItem: 'search' };

  componentDidMount() {
    const { item } = this.props;
    this.setState({ activeItem: item });
  }

  handleItemClick = (e, { name }) => {
    Router.pushRoute(`/${name}`);
  }

  render() {
    const { activeItem } = this.state;

    return (
      <div>
        <Message>
          <Message.Header as="h1">
            Steem MULti Tags Search
            <i>(a.k.a smults)</i>
          </Message.Header>
          <p>
            This web application aims to improve the search quality of
            {' '}
            <a href="https://www.steemit.com" target="_blank" rel="noopener noreferrer">steemit.com</a>
            .
            Up to 5 tags can be searched using the search bar.
            Head over to the
            {' '}
            <Link route="/about"><a>about</a></Link>
            {' '}
            tab for more info.
          </p>
        </Message>
        <Menu pointing secondary>
          <Menu.Item name="search" active={activeItem === 'search'} onClick={this.handleItemClick} />
          <Menu.Item name="about" active={activeItem === 'about'} onClick={this.handleItemClick} />
        </Menu>
      </div>
    );
  }
}

Header.propTypes = {
  item: PropTypes.string.isRequired,
};

export default Header;
