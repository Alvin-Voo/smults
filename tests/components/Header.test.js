import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import routes from '../../routes';

jest.mock('../../routes');

describe('Basic rendering tests ', () => {
  test('should render Header correctly with default "search" tab', () => {
    const wrapper = shallow(<Header item="search" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Header correctly with "about" tab', () => {
    const wrapper = shallow(<Header item="about" />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Router mocks tests', () => {
  beforeEach(() => {
    // Clear all instances and calls for every run
    routes.Router.pushRoute.mockClear();
  });

  test('should route search click handler correctly', () => {
    const wrapper = shallow(<Header item="about" />);
    wrapper.find('MenuItem').at(0).simulate('click', {}, {
      name: 'search',
    });
    expect(routes.Router.pushRoute).toHaveBeenCalledTimes(1);
    expect(routes.Router.pushRoute).toHaveBeenCalledWith('/search');
  });

  test('should route about click handler correctly', () => {
    const wrapper = shallow(<Header item="search" />);
    expect(routes.Router.pushRoute).toHaveBeenCalledTimes(0);
    wrapper.find('MenuItem').at(1).simulate('click', {}, {
      name: 'about',
    });
    expect(routes.Router.pushRoute).toHaveBeenCalledTimes(1);
    expect(routes.Router.pushRoute).toHaveBeenCalledWith('/about');
  });
});
