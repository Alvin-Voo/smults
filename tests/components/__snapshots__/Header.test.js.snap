// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic rendering tests  should render Header correctly with "about" tab 1`] = `
<div>
  <Message>
    <MessageHeader
      as="h1"
    >
      Steem MULti Tags Search
      <i>
        (a.k.a smults)
      </i>
    </MessageHeader>
    <p>
      This web application aims to improve the search quality of
       
      <a
        href="https://www.steemit.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        steemit.com
      </a>
      . Up to 5 tags can be searched using the search bar. Head over to the
       
      <LinkRoutes
        route="/about"
      >
        <a>
          about
        </a>
      </LinkRoutes>
       
      tab for more info.
    </p>
  </Message>
  <Menu
    pointing={true}
    secondary={true}
  >
    <MenuItem
      active={false}
      name="search"
      onClick={[Function]}
    />
    <MenuItem
      active={true}
      name="about"
      onClick={[Function]}
    />
  </Menu>
</div>
`;

exports[`Basic rendering tests  should render Header correctly with default "search" tab 1`] = `
<div>
  <Message>
    <MessageHeader
      as="h1"
    >
      Steem MULti Tags Search
      <i>
        (a.k.a smults)
      </i>
    </MessageHeader>
    <p>
      This web application aims to improve the search quality of
       
      <a
        href="https://www.steemit.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        steemit.com
      </a>
      . Up to 5 tags can be searched using the search bar. Head over to the
       
      <LinkRoutes
        route="/about"
      >
        <a>
          about
        </a>
      </LinkRoutes>
       
      tab for more info.
    </p>
  </Message>
  <Menu
    pointing={true}
    secondary={true}
  >
    <MenuItem
      active={true}
      name="search"
      onClick={[Function]}
    />
    <MenuItem
      active={false}
      name="about"
      onClick={[Function]}
    />
  </Menu>
</div>
`;
